packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx

```diff
@@ -13,17 +13,38 @@ import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import clsx from 'clsx';
 import styles from './styles.module.css';
 
-const Moon = () => <span className={clsx(styles.toggle, styles.moon)} />;
-const Sun = () => <span className={clsx(styles.toggle, styles.sun)} />;
+const Dark = ({icon, style}) => (
+  <span className={clsx(styles.toggle, styles.dark)} style={style}>
+    {icon}
+  </span>
+);
+const Light = ({icon, style}) => (
+  <span className={clsx(styles.toggle, styles.light)} style={style}>
+    {icon}
+  </span>
+);
 
 export default function (props: ComponentProps<typeof Toggle>): JSX.Element {
-  const {isClient} = useDocusaurusContext();
+  const {isClient, siteConfig = {}} = useDocusaurusContext();
+
+  const {
+    themeConfig: {
+      colorMode: {
+        switchConfig: {
+          darkIcon = '🌜',
+          darkIconStyle = {},
+          lightIcon = '🌞',
+          lightIconStyle = {},
+        } = {},
+      } = {},
```

Comment:
We have introduced config/options/theme normalization exactly to avoid such code ^^ we don't want components to handle themselves default values, we want to always provide values to components in the first place

Reply:
Ah yep, did you want me to make a central DEFAULT_CONFIG for docusaurus-theme-classic in the same way you validated algolia's theme config?

Reply:
Not necessarily central, but you can make a `const DEFAULT_COLOR_MODE`